api/api.go

```diff
@@ -34,10 +34,12 @@ func configureExtensionsApi(config *core.Config, router *mux.Router, ctx context
 	api := &ExtensionsApi{
 		core.NewExtensionService(config.Extensions, config.Port),
 		router,
-		newNotifier(ctx),
-		false,
+		make(map[*websocket.Conn]func(status StatusUpdate)),
+		sync.Mutex{},
```

Comment:
Turns out, `sync.Map` is a thing. This will safe us from managing a Mutex ourselves.

Reply:
Oooh! sounds good